src/java.base/share/classes/java/lang/Math.java

```diff
@@ -1346,16 +1347,278 @@ public static long negateExact(long a) {
      * Returns the value of the {@code long} argument,
      * throwing an exception if the value overflows an {@code int}.
      *
-     * @param value the long value
-     * @return the argument as an int
-     * @throws ArithmeticException if the {@code argument} overflows an int
+     * @param value the {@code long} value
+     * @return the argument as an {@code int}
+     * @throws ArithmeticException if the argument overflows an {@code int}
      * @since 1.8
      */
+    @ForceInline
     public static int toIntExact(long value) {
-        if ((int)value != value) {
-            throw new ArithmeticException("integer overflow");
+        int result = (int) value;
+        if ((long) result != value) {
+            throw newArithmeticExceptionOverflow(value);
+        }
+        return result;
+    }
+
+    /**
+     * Returns the value of the {@code double} argument,
+     * throwing an exception if the conversion is inexact.
+     * The method returns iff the argument and the result
```

Comment:
Style suggestion: given the audience of the javadoc, rather than "iff" write out "if and only if" or other full word construct.